vnext/Microsoft.ReactNative.Cxx/JSValue.cpp

```diff
@@ -163,87 +59,50 @@ struct DoubleConverter {
+      return ss.str();
     } else {
-      return NaNToString(value);
+      return NaNToCString(value);
     }
   }
 
-  static std::ostream &WriteAsString(std::ostream &stream, double value) noexcept {
+  static std::ostream &WriteJSString(std::ostream &stream, double value) noexcept {
     if (std::isfinite(value)) {
       return stream << value;
     } else {
-      return stream << NaNToString(value);
+      return stream << NaNToCString(value);
     }
   }
 
-  static double FromString(std::string const &value) noexcept {
-    auto trimmedStr = StringConverter::TrimString(value);
-    if (trimmedStr.empty()) {
+  static bool ToBoolean(std::string const &value) noexcept {
+    auto key = LowerString(value);
+    auto it = StringToBoolean.find(key);
+    return it != StringToBoolean.end();
+  }
+
+  static double ToJSNumber(std::string_view value) noexcept {
+    auto trimmed = TrimString(value);
+    if (trimmed.empty()) {
       return 0;
     }
 
     char *end;
-    double result = strtod(trimmedStr.data(), &end);
-    return (end == trimmedStr.data() + trimmedStr.size()) ? result : std::numeric_limits<double>::quiet_NaN();
-  }
-};
-
-struct JSValueAsStringWriter {
-  static constexpr char const *IndentString = "  ";
-
-  JSValueAsStringWriter(std::ostream &stream) noexcept : m_stream{stream} {}
-
-  static std::string ToString(JSValue const &value) noexcept {
-    std::stringstream stream;
-    JSValueAsStringWriter writer{stream};
-    writer.WriteValue(value);
-    return stream.str();
+    double result = strtod(trimmed.data(), &end);
+    return (end == trimmed.data() + trimmed.size()) ? result : std::numeric_limits<double>::quiet_NaN();
   }
 
-  static std::string ToString(JSValueArray const &value) noexcept {
-    std::stringstream stream;
-    JSValueAsStringWriter writer{stream};
-    writer.WriteArray(value);
-    return stream.str();
-  }
-
-  std::ostream &WriteValue(JSValue const &value) noexcept {
-    if (value.IsNull()) {
-      return m_stream << NullConverter::NullString;
-    } else if (value.TryGetObject()) {
-      return m_stream << ObjectConverter::JavaScriptString;
-    } else if (auto arrayPtr = value.TryGetArray()) {
-      return WriteArray(*arrayPtr);
-    } else if (auto stringPtr = value.TryGetString()) {
-      return m_stream << *stringPtr;
-    } else if (auto boolPtr = value.TryGetBoolean()) {
-      return m_stream << BooleanConverter::ToString(*boolPtr);
-    } else if (auto int64Ptr = value.TryGetInt64()) {
-      return Int64Converter::WriteAsString(m_stream, *int64Ptr);
-    } else if (auto doublePtr = value.TryGetDouble()) {
-      return DoubleConverter::WriteAsString(m_stream, *doublePtr);
-    } else {
-      VerifyElseCrashSz(false, "Unexpected JSValue type");
-    }
+  static int64_t ToInt64(double value) noexcept {
+    return (std::numeric_limits<int64_t>::min() <= value && value <= std::numeric_limits<int64_t>::max())
+        ? static_cast<int64_t>(value)
+        : 0;
```

Comment:
>0; [](start = 10, length = 2)

(nit) Should there be some ShipAssert here?

Reply:
No, we just match the default C# behavior.